### PR TITLE
Vidible embeds in association array SDPA-233

### PR DESCRIPTION
**Pull request by vied12**

### CODE REVIEW

File: scripts/superdesk/editor2/add-embed.ctrl.js [hunk 1]
```diff
@@ -55,21 +53,81 @@ function SdAddEmbedController (embedService, $element, $timeout, $q, _, EMBED_PR
...
+                        pattern: /twitter\.com\/widgets\.js/g,
+                        name: EMBED_PROVIDERS.twitter
+                    },
+                    {
+                        pattern: /www\.youtube\.com/g,
+                        name: EMBED_PROVIDERS.youtube
+                    },
+                    {
+                        pattern: /src="(.*vidible\.tv.*pid=.*.js)/g,
+                        name: EMBED_PROVIDERS.vidible,
```
Comment (ahilles107) [inline scripts/superdesk/editor2/add-embed.ctrl.js]: typo? `vidible`?

File: scripts/superdesk/editor2/add-embed.ctrl.js [hunk 2]
```diff
@@ -55,21 +53,81 @@ function SdAddEmbedController (embedService, $element, $timeout, $q, _, EMBED_PR
...
+                        callback: function(match) {
+                            var url = 'https://' + match[1];
+                            // $http raise a Same Origin Policy error, so we use jquery here
```
Comment (ahilles107) [inline scripts/superdesk/editor2/add-embed.ctrl.js]: Why trick like that? What's wrong wit cors here?

File: scripts/superdesk/editor2/editor.ctrl.js [hunk 3]
```diff
@@ -169,7 +169,8 @@ function SdTextEditorController(_, EMBED_PROVIDERS, $timeout, $element, editor)
                         if (block.blockType === 'embed') {
                             var blockName = block.embedType.trim();
                             // add an id to the image in order to retrieve it in `assocations` field
-                            if (block.embedType === 'Image') {
+                            console.log('association', block.association);
```
Comment (ahilles107) [inline scripts/superdesk/editor2/editor.ctrl.js]: remove debug

File: scripts/superdesk/editor2/add-embed.ctrl.js (hunk 2 above)
Comment (vied12) [inline scripts/superdesk/editor2/add-embed.ctrl.js]: I don't know why $http raises this error and jquery doesn't... maybe `jquery.ajax` change the `origin` value.

**Comment by naomiaro** [comment]
:+1:

File: scripts/superdesk/editor2/add-embed.ctrl.js (hunk 2 above)
Comment (actionless) [inline scripts/superdesk/editor2/add-embed.ctrl.js]: mb use `//` instead of https?

Comment (vied12) [inline scripts/superdesk/editor2/add-embed.ctrl.js]: Should I? https isn't better? :)

Comment (ahilles107) [inline scripts/superdesk/editor2/add-embed.ctrl.js]: if `https` is supported then it's better to use it instead relative url.

**Pull request merged**